### ISSUE TITLE
`Out-String -Stream` unexpectedly does not split multi-line input strings into individual lines too

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#14638]
<sup>Note: Originally submitted as a suggestion, now reclassified as as bug and rewritten accordingly.</sup>

Due to a presumed oversight, input objects to `Out-String -Stream` that happen to be _multi-line_ strings are _not_ split into individual lines, even though doing so is the purpose of the `-Stream` switch - see [this Stack Overflow answer](https://stackoverflow.com/a/65764914/45375) for background information.

## Steps to reproduce

```powershell
# Pass a single 3-line string, whose lines should be emitted individually.
"a`nb`nc" | Out-String -Stream | Should -Be a, b, c
```

Note: As with `Get-Content`, CR, CRLF, and LF newlines should be recognized interchangeably.

## Expected behavior

The test should succeed.

## Actual behavior

The test fails:

```none
Expected @('a', 'b', 'c'), but got 'a b c'
```

That is, the multi-line input string is passed through as-is (Pester converts it to a single-line representation for the sake of a concise error message).

## Environment data

```none
PowerShell Core 7.2.0-preview.2
```

**Comment by iSazonov** [comment]
[From docs for Stream parameter](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/out-string?view=powershell-7.1):
> Indicates that the cmdlet sends _a separate string for each line of an input object_. By default, the strings for each object are accumulated and sent as a single string.

It seems the request follows the docs, yes?

**Comment by iRon7** [comment]
If this request is accepted, I think it should split lines on a common newline definition: e.g. `'\r?\n'` (and not on `[Environment]::NewLine`) as  the data might be interchanged between operating systems as suggested in the related StackOverflow question.

**Comment by mklement0** [comment]
@iSazonov, yes, implementing the proposal would be consistent with the current documentation, though, prompted by @iRon7, a change to the `Out-String` help topic is pending to note the _actual_ current behavior.

@iRon7, good point - I've updated the OP accordingly.

**Comment by iSazonov** [comment]
I would say that this is a _bug_. In this case, we could fix it quickly.

**Comment by mklement0** [comment]
Good point, @iSazonov - can you please relabel the issue?

**Comment by SteveL-MSFT** [comment]
It seems that this behavior is consistent with Windows PowerShell so the docs should be updated to reflect the current behavior otherwise it's a breaking change.  `-Stream` appears to simply output a string for each input object.  A string input object that contains newlines is still a single object.

**Comment by mklement0** [comment]
@SteveL-MSFT

> `-Stream` appears to simply output a string for each input object

It doesn't, and it shouldn't  - please see https://github.com/MicrosoftDocs/PowerShell-Docs/issues/7153#issuecomment-763815304

**Comment by iSazonov** [comment]
> It seems that this behavior is consistent with Windows PowerShell

Windows PowerShell was always very conservative - minor issues was never fixed. I think it was justified. As a result, users are often unpleasantly surprised and use workarounds. If you look at how many similar problems @mklement0  found, it becomes clear how often users are forced to do this.
I strongly vote to fix such issues in 7.3 (skip for next 7.2 LTS) so that users get more clean PowerShell language in 7.4 LTS (I hope it will already replace Windows PowerShell :-) )

**Comment by mklement0** [comment]
Thanks, @iSazonov. 

Yes, the behavior is consistent with Windows PowerShell, but all that means is that the bug is present there too.
While there is buggy / questionable legacy behavior that users indeed either (a) have _taken advantage_ of or (b) have employed workarounds for that _a fix would break_, neither applies in this case:

* The bug results in unequivocally unhelpful behavior. 
* Existing workarounds would _not_ break if we fix this (the workaround is to apply line-splitting manually, which would simple be a (then-unneeded) no-op).

Therefore, I think a fix for 7.2 is perfectly appropriate.

To save others having to follow the link in my previous comment about the intent behind and current behavior of `-Stream`:

* `Out-String` by default emits a single multi-line string that is the formatted string representation of all of its input, as you would see it in the console, including a trailing newline (which arguably shouldn't be added, but that's a separate discussion - see https://github.com/PowerShell/PowerShell/issues/14444).

* If you add `-Stream`, the lines of this string are output one by one - except if multi-line strings happen to be among the input objects, which are _not_ split into their lines (they are output as is); this is the bug at hand.

That is, even the lines that make up the formatting-system's string representation of a _single_ object are output _individually_; e.g.:

```powershell
# Use Out-String -Stream and enclose all resulting lines in [...]
PS> Get-Item / | Out-String -Stream | ForEach-Object { "[$($_.TrimEnd())]" }
[]
[]
[    Directory:]
[]
[]
[Mode                 LastWriteTime         Length Name]
[----                 -------------         ------ ----]
[d--hs-         1/21/2021   9:40 AM                C:\]
[]
[]
```
